src/binder/bind/copy/bind_copy_from.cpp

```diff
@@ -132,6 +132,19 @@ std::unique_ptr<BoundStatement> Binder::bindCopyRelFrom(const parser::Statement&
         expectedColumnTypes, clientContext);
     auto boundSource = bindScanSource(copyStatement.getSource(),
         copyStatement.getParsingOptionsRef(), expectedColumnNames, expectedColumnTypes);
+    expression_vector warningDataExprs;
+    bool ignoreErrors = CopyConstants::DEFAULT_IGNORE_ERRORS;
+    if (boundSource->type == ScanSourceType::FILE) {
```

Comment:
We should make this a general concept for `BoundScanSource`. If source is not of type `File`, you could still have an empty `waringDataExprs`. 

Let's try to avoid having assumptions like `if (boundSource->type == ScanSourceType::FILE)`

Reply:
added virtual `get*()` functions in `BoundBaseScanSource`